File: 백준/Silver/10866. 덱/덱.cc
#include<iostream>
#include<deque>
#include<string>

using namespace std;

int main() {

	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	deque<int> d_q;
	string str;
	int N;
	int num;

	cin >> N;

	for (int i = 0; i < N; i++) {
		cin >> str;

		if (str == "push_front") {
			cin >> num;
			d_q.push_front(num);
		}
		else if (str == "push_back") {
			cin >> num;
			d_q.push_back(num);
		}

		else if (str == "pop_front") {
			if (!d_q.empty()) {
				cout << d_q.front() << '\n';
				d_q.pop_front();
			}
			else {
				cout << -1 << '\n';
			}
		}
		else if (str == "pop_back") {
			if (!d_q.empty()) {
				cout << d_q.back() << '\n';
				d_q.pop_back();
			}
			else {
				cout << -1 << '\n';
			}
		}

		else if (str == "size") {
			cout << d_q.size() << '\n';
		}
		else if (str == "empty") {
			if (d_q.empty()) {
				cout << 1 << '\n';
			}
			else {
				cout << 0 << '\n';
			}
		}
		else if (str == "front") {
			if (!d_q.empty()) {
				cout << d_q.front() << '\n';
			}
			else {
				cout << -1 << '\n';
			}
		}
		else if (str == "back") {
			if (!d_q.empty()) {
				cout << d_q.back() << '\n';
			}
			else {
				cout << -1 << '\n';
			}
		}
	}


	return 0;
}

File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.ArrayDeque;
import java.util.Deque;

class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Deque<Integer> deque = new ArrayDeque<>();
        int N = Integer.parseInt(br.readLine());

        for (int i=1; i<=N; i++) {
            String[] input = br.readLine().split(" ");
            switch (input[0]) {
                case "push_front":
                    deque.addFirst(Integer.parseInt(input[1]));
                    break;
                case "push_back":
                    deque.addLast(Integer.parseInt(input[1]));
                    break;
                case "pop_front":
                    if (deque.isEmpty())
                        bw.write("-1\n");
                    else
                        bw.write(deque.removeFirst()+"\n");
                    break;
                case "pop_back":
                    if (deque.isEmpty())
                        bw.write("-1\n");
                    else
                        bw.write(deque.removeLast()+"\n");
                    break;
                case "size":
                    bw.write(deque.size()+"\n");
                    break;
                case "empty":
                    if (deque.isEmpty())
                        bw.write("1\n");
                    else
                        bw.write("0\n");
                    break;
                case "front":
                    if (deque.isEmpty())
                        bw.write("-1\n");
                    else
                        bw.write(deque.peekFirst()+"\n");
                    break;
                case "back":
                    if (deque.isEmpty())
                        bw.write("-1\n");
                    else
                        bw.write(deque.peekLast()+"\n");
                    break;
            }
        }



        bw.flush();
        bw.close();
    }
}